tests/scripts/all.sh

```diff
@@ -914,6 +924,20 @@ component_build_no_sockets () {
     make CC=gcc CFLAGS='-Werror -Wall -Wextra -O0 -std=c99 -pedantic' lib
 }
 
+component_test_memory_buffer_allocator () {
+    msg "build: default config with memory buffer allocator enabled"
+    scripts/config.pl set MBEDTLS_MEMORY_BUFFER_ALLOC_C
```

Comment:
You also need to enable `MBEDTLS_PLATFORM_MEMORY` (so says CI (https://jenkins-internal.mbed.com/job/mbedtls-pr-multibranch/job/PR-2469-merge/6/execution/node/1035/log/), and it's right, otherwise this run wouldn't be testing much).

Reply:
Addressed in e273ad0.